ledger/participant-integration-api/src/main/scala/platform/store/packagemeta/PackageMetaDataViewImpl.scala

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.packagemeta
+
+import com.daml.daml_lf_dev.DamlLf.Archive
+import com.daml.lf.archive.Decode
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.DottedName
+
+trait PackageMetadata {
+  def interfaceImplementedBy(interface: Ref.Identifier): Set[Ref.Identifier]
+  def interfaceExists(interface: Ref.Identifier): Boolean
+  def templateExists(template: Ref.Identifier): Boolean
+}
+
+trait PackageMetadataView {
+  // TODO DPP-1068: is it okay with this just to move forward? otherwise we need a reset too.
+  def update(archive: Archive): Unit
+
+  /** Computation will be executed with consistent snapshot of underlying data, and result will
+    * be memoized, until re-computation needed (underlying data changed)
+    */
+  def apply[T](computation: PackageMetadata => T): () => T
+}
+
+case class MetadataDefinitions(
+    templates: Set[Ref.Identifier] = Set.empty,
+    interfaces: Set[Ref.Identifier] = Set.empty,
+    interfacesImplementedBy: Map[Ref.Identifier, Set[Ref.Identifier]] = Map.empty,
+) {
+  def ++(additionalDefinitions: MetadataDefinitions): MetadataDefinitions =
+    MetadataDefinitions(
+      templates = templates ++ additionalDefinitions.templates,
+      interfaces = interfaces ++ additionalDefinitions.interfaces,
+      interfacesImplementedBy =
+        additionalDefinitions.interfacesImplementedBy.foldLeft(interfacesImplementedBy) {
+          case (acc, (interface, templates)) =>
+            acc + (interface -> (acc.getOrElse(interface, Set.empty) ++ templates))
+        },
+    )
+}
+
+object MetadataDefinitions {
+  // TODO DPP-1068: [implementation detail] this needs to be implemented by LF utility instead
+  private[packagemeta] def from(archive: Archive): MetadataDefinitions =
+    Decode
+      .decodeArchive(archive, onlySerializableDataDefs = true) // TODO DPP-1068 why true?
```

Comment:
because you only care about templates & interfaces not random crap that is in the daml-lf package.